src/golang/cmd/server/handler/connect_integration.go

```diff
@@ -306,43 +331,139 @@ func ConnectIntegration(
 		return nil, http.StatusInternalServerError, errors.Wrap(err, "Unable to connect integration.")
 	}
 
+	// For those integrations that require asynchronous setup, we spin those up here. When those goroutines are
+	// complete, they write their results back to the config column of their integration entry.
+
 	// TODO(ENG-2523): move base conda env creation outside of ConnectIntegration.
 	if args.Service == shared.Conda {
+		condaDB, err := database.NewDatabase(DB.Config())
```

Comment:
@likawind @hsubbaraj-spiral why do we define a new database here? I copied this from existing code. I would like to document the reason here.

Reply:
> I think both testing and editing an integration should update the exec_state. We can track the source of the change to so we can display "Tested on" vs "Edited on" vs "Created On" timestamps. What do ya'll think? cc: @vsreekanti This ofc means that testing an integration and finding out that it is failing may cause SDK operations that are using that integration to fail. But tbh that seems like proper behavior, since execution is going to fail anyways (it'll just fail earlier now).

Maybe "Last Verified" would make sense here. Or even have "Last Used" and make an entry to the table each time the integration is invoked. That would give a sense of stability even without using the testing integration button.

Reply:
I like the "Last Used" idea, yeah. By making an entry to the table each time, you mean inserting an entry into the integrations table?